test_integration/lib/test/rest_history_test.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:ably_flutter_plugin/ably_flutter_plugin.dart';
+
+import '../test_dispatcher.dart';
+import 'app_key_provision_helper.dart';
+import 'encoders.dart';
+import 'rest_publish_test.dart';
+import 'test_widget_abstract.dart';
+
+class RestHistoryTest extends TestWidget {
+  const RestHistoryTest(TestDispatcherState dispatcher) : super(dispatcher);
+
+  @override
+  TestWidgetState<TestWidget> createState() => RestHistoryTestState();
+}
+
+class RestHistoryTestState extends TestWidgetState<RestHistoryTest> {
+  @override
+  Future<void> test() async {
+    widget.dispatcher.reportLog('init start');
+    final appKey = await provision('sandbox-');
+    final logMessages = <List<String>>[];
+
+    final rest = Rest(
+      options: ClientOptions.fromKey(appKey.toString())
+        ..environment = 'sandbox'
+        ..clientId = 'someClientId'
+        ..logLevel = LogLevel.verbose
+        ..logHandler =
+            ({msg, exception}) => logMessages.add([msg, exception.toString()]),
+    );
+    final channel = rest.channels.get('test');
+    await restMessagesPublishUtil(channel);
+
+    final paginatedResult = await channel.history();
+    final historyDefault = await _history(channel);
+    await Future.delayed(const Duration(seconds: 2));
+
+    final historyLimit4 = await _history(channel, RestHistoryParams(limit: 4));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final historyLimit2 = await _history(channel, RestHistoryParams(limit: 2));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final historyForwardLimit4 = await _history(
+        channel, RestHistoryParams(direction: 'forwards', limit: 4));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final time1 = DateTime.now();
+    //TODO(tiholic) iOS fails without this delay
```

Comment:
Could you please ensure there's an issue for these `TODO` comments relating to the delays required so we can keep track of this technical debt.

Reply:
https://github.com/ably/ably-flutter/issues/86